guardian/editions
# issue-screen: adjust getItemLayout offset based on weather being shown

## Summary

Weather is affecting the item offsets in a `FlatList`. The component does **not** automatically take its height into account to know items positions, so the navigate-to-a-specific-front feature knows some misalignment depending on weather visibility: https://trello.com/c/1Pqbfu26/1068-fix-misalignement-when-opening-a-front

This changeset fixes that by fetching the weather data directly from the `issue-screen` file, but only if the weather should indeed be shown. Otherwise it just does a dummy request. If the weather data is available and is valid, then we assume it will get rendered, then we add the corresponding height in pixel to each item's offset.

This may seem a little ugly and brittle but trying to guess items position without the full layouting algo is the ugly thing in the first place. It's necessary here however, for the sake of performance.

This changeset also illustrate the limits of Apollo, with a bit more boilerplate than what might be necessary. I'd be supportive of moving state such as weather data in a React Context and be done with it :)

## Test Plan

1. Open an article. Slide until you reach the next article. Close the article view. Notice the front has correctly been scrolled to, and aligned.
2. Open Editions list, press a Front. Notice it's correctly scrolled to.
3. Open Settings, switch the weather on/off depending on what it already was. Close settings and make sure this is indeed now showing up (or hiding). Now open the Editions list, press a Front. Notice it's correctly scrolled to:

<img width="300" alt="Screenshot 2019-12-11 at 17 29 28" src="https://user-images.githubusercontent.com/1733570/70644896-cd3a0000-1c3b-11ea-8976-20b94c444132.png">
